jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -607,27 +619,49 @@ public void removeListener(Listener listener)
     public void doStart() throws Exception
     {
         if (LOG.isDebugEnabled())
-            LOG.debug("Scanner start: scanInStart={}, reportExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
-                      _scanInStart, _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
+            LOG.debug("Scanner start: deferScan={}, reportExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
+                _deferInitialScan, _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
 
-        if (_scanInStart)
+        if (!_deferInitialScan)
         {
-            if (_reportExisting)
-            {
-                // if files exist at startup, report them
-                scan();
-                scan(); // scan twice so files reported as stable
-            }
-            else
-            {
-                //just register the list of existing files and only report changes
-                _prevScan = scanFiles();
-            }
+            _scheduler.start();
```

Comment:
Should remove this line, as this is done in `startup()`.

Reply:
Can't remove this one.
As the `_scheduler` doesn't start otherwise in time for the initial scan + call to `schedule()`
Keep in mind that the `super.doStart()` starts the scheduler.

Reply:
For the default `(deferScan == false)` case, we have to start the scheduler manually.

In the original code (on `jetty-10.0.x` HEAD) the `schedule()` call is only done after the `super.doStart()` (which starts the `_scheduler`)